### PR TITLE
[FIX] point_of_sale: correct move amount lot

### PR DESCRIPTION
**Pull request by switch87**
Before this fix, if 2 or more sale lines have the same product with different 
lot-numbers the amount for each move will be set to the total amount of the 
product.

This fix resolves this problem and will only count the amount sold of
each lot.

Resolves https://github.com/odoo/odoo/issues/30006

### CODE REVIEW

File: addons/point_of_sale/models/pos_order.py [hunk 1]
```diff
@@ -841,12 +841,12 @@ def set_pack_operation_lot(self, picking=None):
                 pack_lot_names = [pos_pack.lot_name for pos_pack in pos_pack_lots]
 
                 if pack_lot_names and lots_necessary:
```
Comment (len-odoo) [inline addons/point_of_sale/models/pos_order.py]: `if pos_pack_lots and lots_necessary:`

**Comment by len-odoo** [comment]
Ok, since you removed the `if` I would remove the `# if a lot nr is set...` which is useless but add a comment that the `qty = 1` is the default for serial numbers.

**Comment by switch87** [comment]
@len-odoo, the `# if a lot nr is set...` is actually not useless, but I reworded it to make more clear what it is about.

File: addons/point_of_sale/models/pos_order.py [hunk 2]
```diff
@@ -834,21 +834,18 @@ def set_pack_operation_lot(self, picking=None):
...
-                qty = 0
                 qty_done = 0
                 pack_lots = []
                 pos_pack_lots = PosPackOperationLot.search([('order_id', '=', order.id), ('product_id', '=', move.product_id.id)])
-                pack_lot_names = [pos_pack.lot_name for pos_pack in pos_pack_lots]
 
-                if pack_lot_names and lots_necessary:
-                    for lot_name in list(set(pack_lot_names)):
-                        stock_production_lot = StockProductionLot.search([('name', '=', lot_name), ('product_id', '=', move.product_id.id)])
+                if pack_lots and lots_necessary:
```
[the remark this answers is not in the document]
Comment (switch87) [reply addons/point_of_sale/models/pos_order.py]: No I did not, only tested my initial commit. It should indeed be pos_pack_lots.

**Comment by len-odoo** [comment]
I assume you have retested it by now. 

robodoo r+

**Pull request closed without merging**